### PR TITLE
Added instructions for using a locally built proxy

### LINKED ISSUE #6075: Document how to run with custom built proxy
I'm working on the linkerd2 proxy right now. I built that as a Docker image using `make docker`, however, the docker image produced doesn't work, because it doesn't contain the `/usr/lib/linkerd/linkerd-await` executable invoked by the post start hook. I've looked at the docker file that builds the image in this repo, and it seems to require downloading a proxy release from GitHub. So.... how do we test a custom proxy build with linkerd2? It would be good if the instructions in `BUILD.md` documented that.

### PR DESCRIPTION
**Pull request by jroper**
Fixes #6075

### CODE REVIEW

**Comment by wmorgan** [comment]
Thank you @jroper!

File: BUILD.md [hunk 1]
```diff
@@ -422,6 +422,27 @@ proxy binary:
...
+
+```bash
+./bin/k3d image import cr.l5d.io/linkerd/proxy:dev
+```
+
+Now, to make a pod use your image, add the following annotations to it:
+
+```yaml
+config.linkerd.io/proxy-version: dev
+config.linkerd.io/proxy-await: disabled
```
Comment (olix0r) [inline BUILD.md:443]: I think we can avoid this by updating the devel docker container to use a more recent base image
```suggestion
```

@kleimkuhler will submit a proxy PR

Comment (kleimkuhler) [reply BUILD.md:443]: This was just fixed by linkerd/linkerd2-proxy#999 so all builds rebased on `main` (d87d0dba8afb0912073e9d3b767c252ac7774f88) will have the `linkerd-await` wrapper.

**Review by olix0r — APPROVED** [review]

File: BUILD.md [hunk 2]
```diff
@@ -422,6 +422,26 @@ proxy binary:
 bin/docker-build-proxy
 ```
 
+#### Locally built proxy
+
+If you want to deploy a locally built proxy, you can build it in the [`linkerd2-proxy`](https://github.com/linkerd/linkerd2-proxy) repo by running:
```
Comment (kleimkuhler) [inline BUILD.md:427]: To fix the markdown lint you'll need to make sure this is wrapped at column 80.

**Review by kleimkuhler — APPROVED** [review]
Looks good after fixing the CI markdown lint 👍

File: BUILD.md (hunk 2 above)
Comment (jroper) [reply BUILD.md]: Done.

**Pull request merged**